gateway/ln-gateway/src/federation_manager.rs

```diff
@@ -61,4 +64,41 @@ impl FederationManager {
             next_scid: Arc::new(Mutex::new(INITIAL_SCID)),
         }
     }
+
+    pub async fn add_client(
+        &self,
+        scid: u64,
```

Comment:
this shouldnt need the scid as a parameter, the `FederationManager` should manage it itself right?

Reply:
The reason we need it here for now is because when we restart the gateway and load the clients from the DB, they are already assigned SCIDs. After some more untangling between `Gateway` and the new `LightningManager` + `FederationManager` structs, I think we should be able to ditch this. The main focus of this PR was to keep the structure of the gateway relatively unchanged but simply separate related data into their own structs.

Reply:
Perhaps loading the clients from the db should also be apart of `FederationManager`